cpp/include/stream_compaction.hpp

```diff
@@ -71,6 +72,40 @@ gdf_column apply_boolean_mask(gdf_column const &input,
  * @return gdf_column Column containing copy of all non-null elements of @p input.
  */
 gdf_column drop_nulls(gdf_column const &input);
+
+/**
+* @brief Filter only unique row indices in original order
+*
+* @param key_columns cudf::table containing key columns to check duplicates rows
+* @param keep_first keep first or last entry as unique if duplicate is found
+*
+* @return unique_indices in order
+*/
+//rmm::device_vector<gdf_index_type> get_unique_ordered_indices(const cudf::table& key_columns,
+//                           const bool keep_first);
+
+
+/**
+ * @brief Choices for drop_duplicates API for retainment of duplicate rows
+ */
+typedef enum {
+  KEEP_FIRST = 0,   ///< Keeps first duplicate row
+  KEEP_LAST,        ///< Keeps last  duplicate row 
+  KEEP_FALSE        ///< Don't keep any duplicate rows
```

Comment:
It's not clear to me what this option does. Does this completely remove any rows that aren't unique?

Reply:
Yes I believe so. Instead of *making* all input rows unique by keeping only one copy, this deletes all non-unique input rows, leaving only rows that had no duplicates. @karthikeyann please clarify in the documentation.

Reply:
If that's the case, I think renaming to `KEEP_UNIQUE` would be more informative.

Reply:
Clarified in documentation.
`keep=False` is python argument equivalent for `DataFrame.drop_duplicates` so I followed the same naming here.

Reply:
I'm not a fan of sticking to Python naming if it doesn't make sense. Python is just one of many languages that can wrap libcudf, so it's best to not be married to Python's unique and sometimes confusing naming. As such, I think we should rename to `KEEP_UNIQUE` or something similar.

Reply:
renamed to KEEP_NONE in another suggested review